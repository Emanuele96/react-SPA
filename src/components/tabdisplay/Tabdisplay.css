.Tabdisplay {
  border: red dashed 1px;
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
  align-items: center;
}

.combination {
  height: 100px;
  width: 100%;
  font-size: 1.5rem;
  font-weight: 600;
  cursor: pointer;
  color: white;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  transition: transform 0.3s ease-in-out;
}

.combination:hover {
  transform: scale(1.1);
  transition: transform 0.3s ease-in-out;
}

#combination1 {
  background-color: green;
}

#combination2 {
  background-color: palevioletred;
}

#combination3 {
  background-color: powderblue;
}

#combination4 {
  background-color: orange;
}