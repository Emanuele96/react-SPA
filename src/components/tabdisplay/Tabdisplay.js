import React from 'react';
import TabChoice from '../ui/tabchoice/Tabchoice';
import './Tabdisplay.css';

const Tabdisplay = props => {
  return <div className='Tabdisplay'>{createTabChoices(props)}</div>;
};

const createTabChoices = props => {
  let choices = [];
  for (let i = 1; i <= 4; i++) {
    choices.push(<TabChoice tabNumber={i} onClick={props.onClick} value={i} />);
  }

  return choices;
};

export default Tabdisplay;
