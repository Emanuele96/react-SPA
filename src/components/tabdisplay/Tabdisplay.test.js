import React from 'react';
import renderer from 'react-test-renderer';

import Tabdisplay from './Tabdisplay';



it('renders correctly ', () => {
    const tree = renderer.create(<Tabdisplay/>).toJSON();
    expect(tree).toMatchSnapshot();
});

