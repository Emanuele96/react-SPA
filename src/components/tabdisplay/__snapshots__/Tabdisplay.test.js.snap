// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly  1`] = `
<div
  className="tabdisplay"
>
  <div
    className="tabchoice"
  >
    <button
      className="combination"
      id="combination1"
      value={1}
    >
      Kombinasjon 
      1
    </button>
  </div>
  <div
    className="tabchoice"
  >
    <button
      className="combination"
      id="combination2"
      value={2}
    >
      Kombinasjon 
      2
    </button>
  </div>
  <div
    className="tabchoice"
  >
    <button
      className="combination"
      id="combination3"
      value={3}
    >
      Kombinasjon 
      3
    </button>
  </div>
  <div
    className="tabchoice"
  >
    <button
      className="combination"
      id="combination4"
      value={4}
    >
      Kombinasjon 
      4
    </button>
  </div>
</div>
`;
